src/net.cpp

```diff
@@ -1571,8 +1579,53 @@ void StopMapPort()
 #endif
 
 
+void CConnman::DNSCachesUpdate() const
+{
+    // Only reachable nodes do this, because those are already easy to find.
+    // Unless the node is explicitly set to be non-reachable (-listen=0),
+    // the only way to check if it is reachable is to observe inbound connections.
+    // This check is made every time, because a node may become non-reachable
+    // due to external factors (e.g. local network settings update).
+    if (CountInboundConnections() == 0) return;
```

Comment:
How about a `-dnscloaking` option to force it on/off?

Reply:
Good idea. Will add this config and add the documentation once we get more concept acks.